contracts/src/v0.8/automation/ZKSyncAutomationForwarder.sol

```diff
@@ -31,39 +35,36 @@ contract ZKSyncAutomationForwarder {
    * @param gasAmount is the amount of gas to use in the call
    * @param data is the 4 bytes function selector + arbitrary function data
    * @return success indicating whether the target call succeeded or failed
+   * @return gasUsed the total gas used from this forwarding call
    */
   function forward(uint256 gasAmount, bytes memory data) external returns (bool success, uint256 gasUsed) {
-    if (msg.sender != address(s_registry)) revert();
-    address target = i_target;
-    gasUsed = gasleft();
-    assembly {
-      let g := gas()
-      // Compute g -= PERFORM_GAS_CUSHION and check for underflow
-      if lt(g, PERFORM_GAS_CUSHION) {
-        revert(0, 0)
-      }
-      g := sub(g, PERFORM_GAS_CUSHION)
-      // if g - g//64 <= gasAmount, revert
-      // (we subtract g//64 because of EIP-150)
-      if iszero(gt(sub(g, div(g, 64)), gasAmount)) {
-        revert(0, 0)
-      }
-      // solidity calls check that a contract actually exists at the destination, so we do the same
-      if iszero(extcodesize(target)) {
-        revert(0, 0)
-      }
-      // call with exact gas
-      success := call(gasAmount, target, 0, add(data, 0x20), mload(data), 0, 0)
+    if (msg.sender != address(s_registry)) revert InvalidCaller(msg.sender);
+
+    uint256 g1 = gasleft();
+    bytes memory returnData;
+    // solhint-disable-next-line avoid-low-level-calls
+    (success, returnData) = GAS_BOUND_CALLER.delegatecall{gas: gasAmount}(
+      abi.encodeWithSelector(IGasBoundCaller.gasBoundCall.selector, i_target, gasAmount, data)
+    );
+    uint256 pubdataGasSpent;
+    if (success) {
+      (, pubdataGasSpent) = abi.decode(returnData, (bytes, uint256));
+    }
+    uint256 executionGas = g1 - gasleft();
+    gasUsed = executionGas + pubdataGasSpent;
+    emit GasDetails(executionGas, tx.gasprice, pubdataGasSpent);
+    if (gasUsed > gasAmount) {
+      return (false, gasUsed);
```

Comment:
Will the pubData gas be still charged by zksync in this case, causing the whole tx to fail?